### PR TITLE
[OPPRO-274] Enable velox/prestosql functions reverse(), trim(), md5(), json_array_length() and regexp_extract_all() in Gluten.

### PR DESCRIPTION
**Pull request by lviiii**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Review by zzcclp — APPROVED** [review]
LGTM

File: gluten-core/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java [hunk 1]
```diff
@@ -199,6 +194,23 @@ public static ExpressionNode makeLiteral(Object obj, DataType dataType, Boolean
       } else {
         return makeDecimalLiteral((Decimal) obj);
       }
+    }  else if (dataType instanceof ArrayType) {
+      if (obj == null) {
+        return makeNullLiteral(TypeBuilder.makeDecimal(nullable, 0, 0));
```
Comment (rui-mo) [inline gluten-core/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java:199]: Why make decimal here?

File: gluten-core/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java [hunk 2]
```diff
@@ -199,6 +194,23 @@ public static ExpressionNode makeLiteral(Object obj, DataType dataType, Boolean
...
+        ArrayList<String> list = new ArrayList<>();
+        for (Object array : arrays) {
+          if (array instanceof UTF8String) {
+            list.add(array.toString());
+          } else {
+            throw new UnsupportedOperationException(
+                    String.format("Type not supported: %s.", dataType.toString()));
+          }
+        }
+        return makeListLiteral(list);
```
Comment (rui-mo) [inline gluten-core/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java:212]: @zhejiangxiaomai Would you like to help review the list literal part here?

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala [hunk 3]
```diff
@@ -222,6 +238,14 @@ object ExpressionConverter extends Logging {
             attributeSeq)
         }
         new Murmur3HashTransformer(exprs, expr)
+      case l: StringTrim =>
+        if (l.trimStr != None) {
+          throw new UnsupportedOperationException(s"not supported yet.")
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:243]: Should this be handled by the backend validator?

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala [hunk 4]
```diff
@@ -240,6 +240,7 @@ object ExpressionConverter extends Logging {
         new Murmur3HashTransformer(exprs, expr)
       case l: StringTrim =>
         if (l.trimStr != None) {
+          // todo to be remove when Velox support this argument
```
Comment (zzcclp) [inline gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:243]: it's better use: `// TODO: to be remove when Velox support this argument`

File: gluten-core/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java (hunk 2 above)
Comment (rui-mo) [reply gluten-core/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java:212]: Hi @lviiii, there are already some functions to create List literal such as makeStringListLiteral, makeDoubleListLiteral etc., would you like to take a look?

https://github.com/oap-project/gluten/blob/main/gluten-core/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java#L115-L117

Comment (lviiii) [reply gluten-core/src/main/java/io/glutenproject/substrait/expression/ExpressionBuilder.java]: I've removed the redundant code.

**Review by rui-mo — APPROVED** [review]

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala (hunk 4 above)
Comment (lviiii) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala]: Updated.

**Pull request merged**